bindings/node/test/clientEncryption.test.js

```diff
@@ -186,3 +189,143 @@ describe('ClientEncryption', function() {
+    return this.client.connect().then(() => {
+      this.collection = this.client.db('test').collection('encryption');
+      return this.collection
+        .drop()
+        .catch(err => {
+          if (err.message.match(/ns not found/)) {
+            return;
+          }
+
+          throw err;
+        })
+        .then(() => {
+          this.encryption = new ClientEncryption(this.client, {
+            keyVaultNamespace: 'test.encryption',
+            kmsProviders
+          });
+        });
+    });
+  });
+
+  afterEach(function() {
+    this.encryption = undefined;
+    this.collection = undefined;
+    if (!requirements.SKIP_AWS_TESTS && this.client) {
+      const client = this.client;
+      this.client = undefined;
+      return client.close();
+    }
+  });
+
+  function makeOptions(keyAltNames) {
+    return Object.assign({}, dataKeyOptions, { keyAltNames });
+  }
+
+  describe('errors', function() {
+    [42, 'hello', { keyAltNames: 'foobar' }, /foobar/].forEach(val => {
+      it(`should fail if typeof keyAltNames = ${typeof val}`, function(done) {
+        try {
+          this.encryption.createDataKey('aws', makeOptions(val), () => {
+            done(new Error('expected test to fail'));
+          });
+        } catch (e) {
+          try {
+            expect(e).to.be.an.instanceof(TypeError);
+            done();
+          } catch (_e) {
+            done(_e);
+          }
+        }
+      });
+    });
+
+    [undefined, null, 42, { keyAltNames: 'foobar' }, ['foobar'], /foobar/].forEach(val => {
+      it(`should fail if typeof keyAltNames[x] = ${typeof val}`, function(done) {
+        try {
+          this.encryption.createDataKey('aws', makeOptions([val]), () => {
+            done(new Error('expected test to fail'));
+          });
+        } catch (e) {
+          try {
+            expect(e).to.be.an.instanceof(TypeError);
+            done();
+          } catch (_e) {
+            done(_e);
+          }
+        }
+      });
+    });
+  });
+
+  it('should create a key with keyAltNames', function() {
+    let dataKey;
+    return this.encryption
+      .createDataKey('aws', makeOptions(['foobar']))
+      .then(_dataKey => (dataKey = _dataKey))
+      .then(() => this.collection.findOne({ keyAltNames: 'foobar' }))
+      .then(document => {
+        expect(document)
+          .to.have.property('keyAltNames')
+          .that.includes.members(['foobar']);
+        expect(document).to.have.nested.property('_id.buffer');
+        expect(Buffer.compare(document._id.buffer, dataKey.buffer)).to.equal(0);
+      });
+  });
+
+  it('should create a key with multiple keyAltNames', function() {
+    let dataKey;
+    let doc1;
+    let doc2;
+    return this.encryption
+      .createDataKey('aws', makeOptions(['foobar', 'fizzbuzz']))
+      .then(_dataKey => (dataKey = _dataKey))
+      .then(() => this.collection.findOne({ keyAltNames: 'foobar' }))
```

Comment:
nit: these two can just be a `Promise.all` instead of the redundant `then` blocks and storing to a variable outside the promise chain.